staging/src/k8s.io/apiserver/pkg/audit/context.go

```diff
@@ -39,21 +39,16 @@ type AuditContext struct {
 	RequestAuditConfig RequestAuditConfig
 
 	// Event is the audit Event object that is being captured to be written in
-	// the API audit log. It is set to nil when the request is not being audited.
-	Event *auditinternal.Event
+	// the API audit log.
+	Event auditinternal.Event
```

Comment:
was probably just a clever way to distinguish between enabled and disabled auditing

Reply:
We used to use a pointer since the event gets initialized later in the handler stage, so it was nil for the first part. That's why we needed the separate annotations slice so earlier handlers (e.g. authn) could record annotations.

This change switches away from a pointer so we can start accumulating information (annotations) on the event right away. Sure, we could keep it a pointer and just populate an empty event right away, but I don't see any reason for that.